packages/lesswrong/components/posts/usePost.ts

```diff
@@ -1,15 +1,16 @@
 import { useMulti } from '../../lib/crud/withMulti';
 import { Posts } from '../../lib/collections/posts';
+import { ApolloError } from 'apollo-client';
 
 export const usePostBySlug = ({slug}: {slug: string}):
   {
-    post: PostsPage|null,
+    post: PostsPage,
```

Comment:
This type definition captures the fact that one of the following must be true:
* `post` is null
* `post` is nonnull, loading is `false` and `error` is null
I was hoping to get this type definition to a state where it would instead require one of:
* `post` is nonnull, loading is `false` and `error` is null
* `post` is null, `loading` is true and `error` is null
* `post` is null, `loading` is false and `error` is nonnull
And have type inference magically make it so that if you checked both loading and error, you wouldn't also have to check whether post was null. But I didn't quite get there.